strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/pypi/PypiArtifactController.java

```diff
@@ -0,0 +1,220 @@
+                     HttpServletRequest request)
+    {
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+
+    }
+    @ApiOperation(value = "Used to retrieve simple index")
+    @GetMapping(value = {"/{storageId}/{repositoryId}/simple"}, produces = MediaType.TEXT_HTML_VALUE)
+    public void listSimple(@RepositoryMapping Repository repository,
+                             HttpServletRequest request)
+    {
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+
+    }
+
+    @ApiOperation(value = "Used to retrieve simple project releases")
+    @GetMapping(value = { "/{storageId}/{repositoryId}/simple/{projectName}" })
+    public void listProject(@RepositoryMapping Repository repository,
+                            @PathVariable String projectName,
+                            HttpServletRequest request)
+    {
+
+    }
+
+    @ApiOperation(value = "Entry point for upload, delete, and documentation upload")
+    @PreAuthorize("hasAuthority('ARTIFACTS_DEPLOY')")
+    @PostMapping(value = {"/{storageId}/{repositoryId}"})
+    public ResponseEntity<String> postFunctions(@RepositoryMapping Repository repository,
+                              @RequestParam(value = ":action") String Action,
+                              @RequestParam(value = "protocol_version") Integer ProtocolVersion,
+                              @RequestParam(value = "md5_digest", required = false) String MD5,
+                              @RequestParam(value = "name", required = false) String Name,
+                              @RequestParam(value = "content", required = false) MultipartFile content,
+                              @RequestParam(value = "gpg_signature", required = false) MultipartFile GPGSig,
+                              @RequestParam(value = "version", required = false) String Version,
+                              @RequestParam(value = "filetype", required = false) String FileType,
+                              HttpServletRequest request)
+    {
+        if(Action.equals("file_upload"))
+            return upload(repository,content,GPGSig,Name,Version,MD5,FileType);
+        if(Action.equals("remove_pkg"))
+            return delete(repository,Name,Version);
+
+
+        return ResponseEntity.badRequest().body("Invalid action");
+    }
+
+    public boolean packageNameCorrect(String Name,
+                                      String Version,
+                                      String FileType,
+                                      String Filename)
+    {
+        StringBuilder builder = new StringBuilder(Name);
+
+        builder.append("-").append(Version).append(FileType.equals("bdist_wheel") ? ".whl":".tar.gz");
+
+        return Filename.equals(builder.toString());
+    }
+
+    private ResponseEntity<String> upload(Repository repository,
```

Comment:
This logic should not be implemented in controller. Controllers should be only contains interaction between request service layers and response. So uploading logic should be extracted to somewhere else. This way controller will be cleaner, more neat